src/domain/bitcoin/onchain/index.ts

```diff
@@ -1,16 +1,32 @@
-import { InvalidScanDepthAmount } from "@domain/errors"
+import { InvalidOnChainAddress, InvalidScanDepthAmount } from "@domain/errors"
 
 export * from "./errors"
 export * from "./tx-filter"
 export * from "./tx-decoder"
 
+export const checkedToOnChainAddress = (
```

Comment:
Should take an additional `network` arg and only return the addresses that are valid for the specific network.